src/Components/Blazor/Blazor/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -91,7 +92,9 @@ private void CreateServiceProvider()
             services.AddSingleton<IJSRuntime, MonoWebAssemblyJSRuntime>();
 
             services.AddSingleton<IUriHelper>(WebAssemblyUriHelper.Instance);
-            services.AddSingleton<HttpClient>(s =>
+            services.AddSingleton<ComponentEnvironment, WebAssemblyComponentEnvironment>();
```

Comment:
We might want to make this Scoped for consistency with server-side scenarios, but it doesn't really make a difference.

Reply:
Is it necessary to remove the `HttpClient` service?

Will this alone stop people from putting `@inject HttpClient Http` in their components? We have this all over our samples and tutorials and definitely don't want to break it if at all possible. Can there be an `HttpClient` service that gets the value from the `ComponentEnvironment`?

Reply:
OK I see it looks like you are registering services that do get `IJSRuntime` (etc.) from the component environment, so it's not breaking for those :)

However maybe it is breaking for `HttpClient`. Is it problematic to retain this service? Do we need to add it to the remote/prerendering cases too for consistency?

Reply:
i didn't remove httpclient. The diff is doing weird things I believe.